### PR TITLE
fix binaryen_root detection

### PR DESCRIPTION
**Pull request by kripken**
See #5340.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -226,7 +226,14 @@ def new(*args):
...
     llvm_root = os.path.dirname(find_executable('llvm-dis') or '/usr/bin/llvm-dis')
     config_file = config_file.replace('\'{{{ LLVM_ROOT }}}\'', repr(llvm_root))
-    binaryen_root = os.path.dirname(find_executable('asm2wasm') or '') # if we don't find it, we'll use the port
+    # binaryen is a little different: it's root is not the dir containing the
+    # executable, but the dir of binaryen itself, as we need access to other things
+    # in it (like scripts/).
+    binaryen_root = os.path.dirname(os.path.dirname(find_executable('asm2wasm') or '')) # if we don't find it, we'll use the port
+    if binaryen_root:
+      # check if it in fact contains all of binaryen
+      if not os.path.exists(os.path.join('binaryen_root', 'scripts')):
```
Comment (sbc100) [inline tools/shared.py]: The quotes shouldn't be around binaryen_root here should they?

Comment (kripken) [inline tools/shared.py]: oops, yeah, that's definitely wrong.

this should really have a test, which would have caught that, i'll get back to this with that when i can.

**Comment by kripken** [comment]
I changed my mind on this. Actually, we shouldn't look for binaryen in first run setup, since we will get binaryen from ports automatically anyhow if the user does not set a location for it, and if we find a local version, it is not guaranteed to be compatible (unlike ports), so best to make the user explicitly set a non-ports version if they want that, instead of us implicitly finding a bad one.

**Comment by juj** [comment]
Looks good to me. (Emscripten SDK always explicitly writes such a `.emscripten` file that points to a precompiled Binaryen, so that path is not affected)

**Pull request merged**